service/lib/agama/autoyast/profile_checker.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2025] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "agama/autoyast/profile_description"
+
+module Agama
+  module AutoYaST
+    # This class checks an AutoYaST profile and determines which unsupported elements are used.
+    #
+    # It does not report unknown elements.
```

Comment:
So what happens with unknown elements? Are they silently ignored now or does a different part of Agama report them?

Reply:
I decided to ignore them by now and report only the status of the known elements. The idea is to extend the `compat` file to cover all the AutoYaST elements, stating whether it is unsupported, supported or planned.